plotly/tools.py

```diff
@@ -3442,3 +3694,226 @@ def get_dendrogram_traces(self, X, colorscale):
+            min_text_color = '#FFFFFF'
+            max_text_color = '#000000'
+        elif self.colorscale in colorscales_opp and self.reversescale:
+            min_text_color = '#FFFFFF'
+            max_text_color = '#000000'
+        elif self.colorscale in colorscales_opp:
+            min_text_color = '#000000'
+            max_text_color = '#FFFFFF'
+        elif isinstance(self.colorscale, list):
+            if 'rgb' in self.colorscale[0][1]:
+                min_col = map(int,
+                              self.colorscale[0][1].strip('rgb()').split(','))
+                max_col = map(int,
+                              self.colorscale[-1][1].strip('rgb()').split(','))
+            elif '#' in self.colorscale[0][1]:
+                    min_col = FigureFactory._hex_to_rgb(self.colorscale[0][1])
+                    max_col = FigureFactory._hex_to_rgb(self.colorscale[-1][1])
+            else:
+                min_col = [255, 255, 255]
+                max_col = [255, 255, 255]
+
+            if (min_col[0]*0.299 + min_col[1]*0.587 + min_col[2]*0.114) > 186:
+                min_text_color = '#000000'
+            else:
+                min_text_color = '#FFFFFF'
+            if (max_col[0]*0.299 + max_col[1]*0.587 + max_col[2]*0.114) > 186:
+                max_text_color = '#000000'
+            else:
+                max_text_color = '#FFFFFF'
+        else:
+            min_text_color = '#000000'
+            max_text_color = '#000000'
+        return min_text_color, max_text_color
+
+    def get_min_max(self):
+        '''
+        Get min and max vals of z matrix
+
+        :rtype (float, float) z_min, z_max: min and max vals from z matrix
+        '''
+        if _numpy_imported and isinstance(self.z, np.ndarray):
+            z_min = np.amin(self.z)
+            z_max = np.amax(self.z)
+        else:
+            z_min = min(min(self.z))
+            z_max = max(max(self.z))
+        return z_min, z_max
+
+    def make_annotations(self):
+        '''
+        Get annotations for each cell of the heatmap with graph_objs.Annotation
+
+        :rtype (list) annotations: list of annotations for each cell of the
+            heatmap
+        '''
+        from plotly.graph_objs import graph_objs
+        min_text_color, max_text_color = _AnnotatedHeatmap.get_text_color(self)
+        z_min, z_max = _AnnotatedHeatmap.get_min_max(self)
+        annotations = []
+        for n, row in enumerate(self.z):
+            for m, val in enumerate(row):
+                annotations.append(
+                    graph_objs.Annotation(
+                        text=str(self.annotation_text[n][m]),
+                        x=self.x[m],
+                        y=self.y[n],
+                        xref='x1',
+                        yref='y1',
+                        font=dict(color=min_text_color if val <
+                                  (z_max+z_min) / 2
+                                  else max_text_color),
```

Comment:
:cow2: it's better to define a variable in this case, imo:

`font_color = min_text_color if val < (z_max+z_min) / 2 else max_text_color`

And then just set font to:

`font=dict(color=font_color)`